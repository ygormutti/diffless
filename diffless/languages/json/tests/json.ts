import { announceHtml, fixtureDocument } from '../../../tests/test-util';
import { jsonDiff, jsonLexicalDiff } from '../index';

describe('JSON support', () => {
    const mapReorderleft = fixtureDocument('single_map_reorder/before.json');
    const mapReorderRight = fixtureDocument('single_map_reorder/after.json');

    describe('jsonLexicalDiff', () => {
        it('should match snapshot', () => {
            const diff = jsonLexicalDiff(mapReorderleft, mapReorderRight);

            expect(diff.edits).toMatchSnapshot();
            announceHtml(mapReorderleft, mapReorderRight, diff.edits, 'json_lexical_1');
        });
    });

    describe('jsonDiff', () => {
        it('should match snapshot', () => {
            const diff = jsonDiff(mapReorderleft, mapReorderRight);

            expect(diff.edits).toMatchSnapshot();
            announceHtml(mapReorderleft, mapReorderRight, diff.edits, 'json_1');
        });

        it('should match snapshot', () => {
            const left = fixtureDocument('textual_changes/before.json');
            const right = fixtureDocument('textual_changes/after.json');

            const diff = jsonDiff(left, right);

            expect(diff.edits).toMatchSnapshot();
            announceHtml(left, right, diff.edits, 'textual_vs_json_1');
        });
    });
});
