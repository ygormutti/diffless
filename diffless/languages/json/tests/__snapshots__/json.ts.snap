// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JSON support jsonDiff should match snapshot 1`] = `
Array [
  Object {
    "left": [2:5; 2:15) @ file://diffless/tests/fixtures/single_map_reorder/before.json,
    "level": "Lexical",
    "operation": "Move",
    "right": [14:3; 14:13) @ file://diffless/tests/fixtures/single_map_reorder/after.json,
  },
  Object {
    "left": [2:16; 2:35) @ file://diffless/tests/fixtures/single_map_reorder/before.json,
    "level": "Lexical",
    "operation": "Move",
    "right": [14:14; 14:33) @ file://diffless/tests/fixtures/single_map_reorder/after.json,
  },
  Object {
    "left": [2:35; 2:36) @ file://diffless/tests/fixtures/single_map_reorder/before.json,
    "level": "Lexical",
    "operation": "Delete",
    "right": undefined,
  },
  Object {
    "left": undefined,
    "level": "Lexical",
    "operation": "Add",
    "right": [13:4; 13:5) @ file://diffless/tests/fixtures/single_map_reorder/after.json,
  },
  Object {
    "left": [2:5; 2:35) @ file://diffless/tests/fixtures/single_map_reorder/before.json,
    "level": "Textual",
    "operation": "Move",
    "right": [14:3; 14:33) @ file://diffless/tests/fixtures/single_map_reorder/after.json,
  },
  Object {
    "left": [1:1; 2:5) @ file://diffless/tests/fixtures/single_map_reorder/before.json,
    "level": "Textual",
    "operation": "Delete",
    "right": undefined,
  },
  Object {
    "left": [2:35; 3:3) @ file://diffless/tests/fixtures/single_map_reorder/before.json,
    "level": "Textual",
    "operation": "Delete",
    "right": undefined,
  },
  Object {
    "left": [4:1; 4:5) @ file://diffless/tests/fixtures/single_map_reorder/before.json,
    "level": "Textual",
    "operation": "Delete",
    "right": undefined,
  },
  Object {
    "left": [5:1; 5:7) @ file://diffless/tests/fixtures/single_map_reorder/before.json,
    "level": "Textual",
    "operation": "Delete",
    "right": undefined,
  },
  Object {
    "left": [6:1; 6:7) @ file://diffless/tests/fixtures/single_map_reorder/before.json,
    "level": "Textual",
    "operation": "Delete",
    "right": undefined,
  },
  Object {
    "left": [7:1; 7:7) @ file://diffless/tests/fixtures/single_map_reorder/before.json,
    "level": "Textual",
    "operation": "Delete",
    "right": undefined,
  },
  Object {
    "left": [8:1; 8:5) @ file://diffless/tests/fixtures/single_map_reorder/before.json,
    "level": "Textual",
    "operation": "Delete",
    "right": undefined,
  },
  Object {
    "left": [9:1; 9:5) @ file://diffless/tests/fixtures/single_map_reorder/before.json,
    "level": "Textual",
    "operation": "Delete",
    "right": undefined,
  },
  Object {
    "left": [10:1; 10:7) @ file://diffless/tests/fixtures/single_map_reorder/before.json,
    "level": "Textual",
    "operation": "Delete",
    "right": undefined,
  },
  Object {
    "left": [11:1; 11:7) @ file://diffless/tests/fixtures/single_map_reorder/before.json,
    "level": "Textual",
    "operation": "Delete",
    "right": undefined,
  },
  Object {
    "left": [12:1; 12:7) @ file://diffless/tests/fixtures/single_map_reorder/before.json,
    "level": "Textual",
    "operation": "Delete",
    "right": undefined,
  },
  Object {
    "left": [13:7; 13:10) @ file://diffless/tests/fixtures/single_map_reorder/before.json,
    "level": "Textual",
    "operation": "Delete",
    "right": undefined,
  },
  Object {
    "left": [14:3; 14:6) @ file://diffless/tests/fixtures/single_map_reorder/before.json,
    "level": "Textual",
    "operation": "Delete",
    "right": undefined,
  },
  Object {
    "left": undefined,
    "level": "Textual",
    "operation": "Add",
    "right": [1:1; 2:1) @ file://diffless/tests/fixtures/single_map_reorder/after.json,
  },
  Object {
    "left": undefined,
    "level": "Textual",
    "operation": "Add",
    "right": [12:5; 13:1) @ file://diffless/tests/fixtures/single_map_reorder/after.json,
  },
  Object {
    "left": undefined,
    "level": "Textual",
    "operation": "Add",
    "right": [13:3; 13:5) @ file://diffless/tests/fixtures/single_map_reorder/after.json,
  },
]
`;

exports[`JSON support jsonDiff should match snapshot 2`] = `
Array [
  Object {
    "left": [6:17; 6:24) @ file://diffless/tests/fixtures/textual_changes/before.json,
    "level": "Lexical",
    "operation": "Move",
    "right": [7:11; 7:18) @ file://diffless/tests/fixtures/textual_changes/after.json,
  },
  Object {
    "left": [7:25; 7:26) @ file://diffless/tests/fixtures/textual_changes/before.json,
    "level": "Lexical",
    "operation": "Move",
    "right": [7:18; 7:19) @ file://diffless/tests/fixtures/textual_changes/after.json,
  },
  Object {
    "left": [8:17; 8:30) @ file://diffless/tests/fixtures/textual_changes/before.json,
    "level": "Lexical",
    "operation": "Move",
    "right": [8:11; 8:24) @ file://diffless/tests/fixtures/textual_changes/after.json,
  },
  Object {
    "left": [6:24; 6:25) @ file://diffless/tests/fixtures/textual_changes/before.json,
    "level": "Lexical",
    "operation": "Delete",
    "right": undefined,
  },
  Object {
    "left": [7:17; 7:25) @ file://diffless/tests/fixtures/textual_changes/before.json,
    "level": "Lexical",
    "operation": "Delete",
    "right": undefined,
  },
  Object {
    "left": [8:30; 8:31) @ file://diffless/tests/fixtures/textual_changes/before.json,
    "level": "Lexical",
    "operation": "Delete",
    "right": undefined,
  },
  Object {
    "left": undefined,
    "level": "Lexical",
    "operation": "Add",
    "right": [6:79; 6:80) @ file://diffless/tests/fixtures/textual_changes/after.json,
  },
  Object {
    "left": [6:9; 6:17) @ file://diffless/tests/fixtures/textual_changes/before.json,
    "level": "Textual",
    "operation": "Move",
    "right": [7:1; 7:9) @ file://diffless/tests/fixtures/textual_changes/after.json,
  },
  Object {
    "left": [6:17; 6:23) @ file://diffless/tests/fixtures/textual_changes/before.json,
    "level": "Textual",
    "operation": "Move",
    "right": [7:11; 7:17) @ file://diffless/tests/fixtures/textual_changes/after.json,
  },
  Object {
    "left": [7:24; 8:1) @ file://diffless/tests/fixtures/textual_changes/before.json,
    "level": "Textual",
    "operation": "Move",
    "right": [7:17; 8:1) @ file://diffless/tests/fixtures/textual_changes/after.json,
  },
  Object {
    "left": [8:17; 8:30) @ file://diffless/tests/fixtures/textual_changes/before.json,
    "level": "Textual",
    "operation": "Move",
    "right": [8:11; 8:24) @ file://diffless/tests/fixtures/textual_changes/after.json,
  },
  Object {
    "left": [9:5; 9:7) @ file://diffless/tests/fixtures/textual_changes/before.json,
    "level": "Textual",
    "operation": "Move",
    "right": [9:1; 9:3) @ file://diffless/tests/fixtures/textual_changes/after.json,
  },
  Object {
    "left": [10:13; 10:15) @ file://diffless/tests/fixtures/textual_changes/before.json,
    "level": "Textual",
    "operation": "Move",
    "right": [9:9; 9:11) @ file://diffless/tests/fixtures/textual_changes/after.json,
  },
  Object {
    "left": [2:1; 2:3) @ file://diffless/tests/fixtures/textual_changes/before.json,
    "level": "Textual",
    "operation": "Delete",
    "right": undefined,
  },
  Object {
    "left": [3:1; 3:5) @ file://diffless/tests/fixtures/textual_changes/before.json,
    "level": "Textual",
    "operation": "Delete",
    "right": undefined,
  },
  Object {
    "left": [4:1; 4:7) @ file://diffless/tests/fixtures/textual_changes/before.json,
    "level": "Textual",
    "operation": "Delete",
    "right": undefined,
  },
  Object {
    "left": [5:1; 5:7) @ file://diffless/tests/fixtures/textual_changes/before.json,
    "level": "Textual",
    "operation": "Delete",
    "right": undefined,
  },
  Object {
    "left": [5:24; 6:9) @ file://diffless/tests/fixtures/textual_changes/before.json,
    "level": "Textual",
    "operation": "Delete",
    "right": undefined,
  },
  Object {
    "left": [6:23; 7:24) @ file://diffless/tests/fixtures/textual_changes/before.json,
    "level": "Textual",
    "operation": "Delete",
    "right": undefined,
  },
  Object {
    "left": [8:1; 8:17) @ file://diffless/tests/fixtures/textual_changes/before.json,
    "level": "Textual",
    "operation": "Delete",
    "right": undefined,
  },
  Object {
    "left": [8:30; 9:5) @ file://diffless/tests/fixtures/textual_changes/before.json,
    "level": "Textual",
    "operation": "Delete",
    "right": undefined,
  },
  Object {
    "left": [9:85; 10:1) @ file://diffless/tests/fixtures/textual_changes/before.json,
    "level": "Textual",
    "operation": "Delete",
    "right": undefined,
  },
  Object {
    "left": [10:15; 11:1) @ file://diffless/tests/fixtures/textual_changes/before.json,
    "level": "Textual",
    "operation": "Delete",
    "right": undefined,
  },
  Object {
    "left": [12:1; 12:5) @ file://diffless/tests/fixtures/textual_changes/before.json,
    "level": "Textual",
    "operation": "Delete",
    "right": undefined,
  },
  Object {
    "left": [13:1; 13:5) @ file://diffless/tests/fixtures/textual_changes/before.json,
    "level": "Textual",
    "operation": "Delete",
    "right": undefined,
  },
  Object {
    "left": [14:1; 14:7) @ file://diffless/tests/fixtures/textual_changes/before.json,
    "level": "Textual",
    "operation": "Delete",
    "right": undefined,
  },
  Object {
    "left": [15:1; 15:7) @ file://diffless/tests/fixtures/textual_changes/before.json,
    "level": "Textual",
    "operation": "Delete",
    "right": undefined,
  },
  Object {
    "left": [16:1; 16:7) @ file://diffless/tests/fixtures/textual_changes/before.json,
    "level": "Textual",
    "operation": "Delete",
    "right": undefined,
  },
  Object {
    "left": [17:1; 17:7) @ file://diffless/tests/fixtures/textual_changes/before.json,
    "level": "Textual",
    "operation": "Delete",
    "right": undefined,
  },
  Object {
    "left": [17:23; 18:5) @ file://diffless/tests/fixtures/textual_changes/before.json,
    "level": "Textual",
    "operation": "Delete",
    "right": undefined,
  },
  Object {
    "left": [19:1; 19:3) @ file://diffless/tests/fixtures/textual_changes/before.json,
    "level": "Textual",
    "operation": "Delete",
    "right": undefined,
  },
  Object {
    "left": [20:1; 20:3) @ file://diffless/tests/fixtures/textual_changes/before.json,
    "level": "Textual",
    "operation": "Delete",
    "right": undefined,
  },
  Object {
    "left": [20:15; 21:8) @ file://diffless/tests/fixtures/textual_changes/before.json,
    "level": "Textual",
    "operation": "Delete",
    "right": undefined,
  },
  Object {
    "left": undefined,
    "level": "Textual",
    "operation": "Add",
    "right": [5:18; 6:1) @ file://diffless/tests/fixtures/textual_changes/after.json,
  },
  Object {
    "left": undefined,
    "level": "Textual",
    "operation": "Add",
    "right": [6:79; 7:1) @ file://diffless/tests/fixtures/textual_changes/after.json,
  },
  Object {
    "left": undefined,
    "level": "Textual",
    "operation": "Add",
    "right": [8:24; 9:1) @ file://diffless/tests/fixtures/textual_changes/after.json,
  },
  Object {
    "left": undefined,
    "level": "Textual",
    "operation": "Add",
    "right": [9:11; 10:1) @ file://diffless/tests/fixtures/textual_changes/after.json,
  },
  Object {
    "left": undefined,
    "level": "Textual",
    "operation": "Add",
    "right": [16:17; 17:1) @ file://diffless/tests/fixtures/textual_changes/after.json,
  },
]
`;

exports[`JSON support jsonLexicalDiff should match snapshot 1`] = `
Array [
  Object {
    "left": [2:5; 2:15) @ file://diffless/tests/fixtures/single_map_reorder/before.json,
    "level": "Lexical",
    "operation": "Move",
    "right": [14:3; 14:13) @ file://diffless/tests/fixtures/single_map_reorder/after.json,
  },
  Object {
    "left": [2:16; 2:35) @ file://diffless/tests/fixtures/single_map_reorder/before.json,
    "level": "Lexical",
    "operation": "Move",
    "right": [14:14; 14:33) @ file://diffless/tests/fixtures/single_map_reorder/after.json,
  },
  Object {
    "left": [2:35; 2:36) @ file://diffless/tests/fixtures/single_map_reorder/before.json,
    "level": "Lexical",
    "operation": "Delete",
    "right": undefined,
  },
  Object {
    "left": undefined,
    "level": "Lexical",
    "operation": "Add",
    "right": [13:4; 13:5) @ file://diffless/tests/fixtures/single_map_reorder/after.json,
  },
]
`;
